client/settings/settings-manager/index.js

```diff
@@ -94,31 +93,30 @@ const SettingsManager = () => {
 	const {
 		featureFlags: { upe: isUPEEnabled },
 	} = useContext( WCPaySettingsContext );
-	const { isLoading } = useSettings();
 
 	return (
 		<SettingsLayout>
 			<SettingsSection Description={ GeneralSettingsDescription }>
-				<LoadableBlock isLoading={ isLoading } numLines={ 20 }>
```

Comment:
This is interesting. TIL.

The `useSettings` hook change causes the entire `<SettingsManager>` and `<SettingsLayout>` to re-render. I would imagine only `<LoadableBlock>` gets re-rendered since that's the only component consuming the `useSettings` hook. This looks like the hook in a component causes a re-render on the component. I am curious if this is only for `useSettings` hook or does this apply to all hooks.

Does this mean we should always move component that consumes hook into its own component? Case by case maybe?

Reply:
I think it goes on a case-by-case basis.
The main issue is that the `useSettings` hook uses `useSelect` to return the `settings` attribute (which represent all the setting values).
Every time the settings get updated, the `setting` `useSelect` will run, which will cause `setting` to change, which will cause a re-render.

This is just one little trick that leverages `children` to prevent the whole component tree from re-rendering.